### ISSUE TITLE
Heading name does not consider content

### ISSUE THREAD

**Issue by Jym77** [w3c/html-aam#280]
As far as I can follow, the Accessible Name computation of heading elements (`<h1>`, …) follows [Section and Grouping Element Accessible Name Computation](https://w3c.github.io/html-aam/#section-and-grouping-element-accessible-name-computation) which is:
1. `aria-label` or `aria-labelledby`
2. `title`
3. empty name

Thus, an element like `<h1>Accessible names are cool</h1>` has neither `aria-label`, `aria-labelledby` nor `title` attribute and will have no accessible name. It seems that it should get an accessible name from content (and that is what UA and AT are doing as far as I can tell).

It make sense that other Section elements (e.g. `<body>`) don't get their name from content as it would get somewhat crazy… But for `<h1>`-`<h6>` element, it looks like a specific case should be added, allowing name from content.

**Comment by JAWS-test** [comment]
This also applies to  Grouping Content like `<p>`, `<li>`, `<dt>`, `<dd>`...

**Comment by Jym77** [comment]
True. But I'm rather happy that a `<p>` element does not get its full content as an accessible name…

Actually, a `<p>` element has no default  role, so does not have a role that allow name from content (I guess) and thus step 2F of the accessible name computation should be skipped…

`<h*>` elements have a default role of `heading` which does allow name from content.

**Comment by JAWS-test** [comment]
`<li>`, `<dt>`, `<dd>` have a default role and `<p>` will get the role `paragraph`.

A distinction may be made between content in the element and labeling of the element. For example, in Chrome labeling these elements with aria-label, JAWS only outputs the text content when reading with a virtual cursor (and not the aria-label), only the aria-label for tab navigation (and not the text content). Contrary to the specification, the title attribute is never output as a label, but only as a description while tab navigation and not at all when reading. See https://codepen.io/jaws-test/pen/gOpbOdw

I am not sure whether the specification is inadequate here, or whether Chrome and JAWS are not implementing the specification correctly or whether the different output of name and content is so desired depending on the navigation mode. At least I think it's very confusing ...

**Comment by scottaohara** [comment]
@Jym77 appreciate the issue.  we need to update this section to add some clarity as some elements that are considered grouping or sectioning elements should allow name from contents (e.g. headings), while others (`body` and `div` for example) should not allow naming at all.

**Comment by JAWS-test** [comment]
Related: https://github.com/w3c/aria/issues/1054, https://github.com/w3c/html-aam/issues/160
